model_analyzer/record/types/inter_token_latency_avg.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import total_ordering
+
+from model_analyzer.record.record import DecreasingRecord
+
+
+@total_ordering
+class InterTokenLatencyAvg(DecreasingRecord):
+    """
+    A record for perf_analyzer Inter token latency metric
+    """
+
+    tag = "inter_token_latency_avg"
```

Comment:
Is there not a way to code these up such that we only need one base class and 6 derived classes for InterTokenLatency?

Reply:
There probably is and that would save us a little bit of code since the __lt__, __eq__, __add__, __sub__ wouldn't need to be replicated, but everything else in the class is still required or am I missing something?

Reply:
I think in my mind this was also going to collapse the number of files and comments, and that it would end up parameterized where the tag and header were passed in to a template or something. I'm fine with how this is though.